packages/lesswrong/lib/scrollUtils.ts

```diff
@@ -0,0 +1,246 @@
+    });
+
+    // Clear currentScrollFocus if it's still pointing to this instance
+    if (currentScrollFocus && currentScrollFocus.cleanup === ref.cleanup) {
+      currentScrollFocus = null;
+    }
+  };
+
+  setTimeout(ref.cleanup, 5000);
+
+  currentScrollFocus = {
+    cleanup: ref.cleanup,
+  };
+}
+
+/**
+ * Simplified version of `scrollFocusOnQueryComment` to run as a standalone script as soon
+ * as the page loads, to speed up interactivity.
+ */
+export const preloadScrollToCommentScript = `<script>
+  try {
+    function prettyScrollTo(element) {
+      const rect = element.getBoundingClientRect();
+      const offset = 25; // See commentIdToLandmark
+      const elementY = rect.top + offset;
+      const scrollPosition = elementY + window.scrollY - (window.innerHeight / 5) + 1;
+      window.scrollTo({ top: scrollPosition });
+    }
+
+    // Function to scroll to the comment specified in the query parameter
+    function scrollFocusOnQueryComment() {
+      const urlParams = new URLSearchParams(window.location.search);
+      let commentId = urlParams.get("commentId");
+
+      if (!commentId) {
+        const hash = window.location.hash;
+        if (hash.startsWith("#")) {
+          commentId = hash.substring(1);
+        }
+      }
+
+      const element = document.getElementById(commentId);
+      if (!element) {
+        return;
+      }
+
+      prettyScrollTo(element);
+
+      const ref = {
+        cleanup: function() {}
+      };
+
+      const observer = new MutationObserver(function() {
+        // Check if the element is still in the DOM
+        if (!element.isConnected) {
+          ref.cleanup();
+          return;
+        }
+
+        prettyScrollTo(element);
+      });
+
+      observer.observe(document.body, {
+        attributes: true,
+        childList: true,
+        subtree: true
+      });
+
+      const userEventListener = function() {
+        ref.cleanup();
+      };
+
+      // Event listeners for user-initiated actions
+      ["mousedown", "keydown", "wheel", "touchstart"].forEach(function(eventType) {
+        window.addEventListener(eventType, userEventListener, { passive: true });
+      });
+
+      ref.cleanup = function() {
+        observer.disconnect();
+        ["mousedown", "keydown", "wheel", "touchstart"].forEach(function(eventType) {
+          window.removeEventListener(eventType, userEventListener);
+        });
+        window.killPreloadScroll = null;
+      };
+
+      // Expose the cleanup function on the window object so the proper version in scrollUtils can take over
+      window.killPreloadScroll = ref.cleanup;
+
+      setTimeout(ref.cleanup, 5000);
+    }
+
+    scrollFocusOnQueryComment();
+  } catch (e) {
+    console.error(e)
+    if (${!isProduction}) { // Note: condition will be constant in rendered html
+      alert("Error in preloaded scrollFocusOnQueryComment script, see renderPage.tsx (this error will only appear on dev).");
+    }
```

Comment:
Because this is just a string it will be prone to breaking silently, so I'm raising an alert on dev (but not prod) if there is an error here

Reply:
A thing you could do is put this into its own `.js` file with a `// @ts-check` at the top (so that it reports type errors), and then load the contents once on server startup with e.g. `fs.readFile` for later use in `buildSSRBody`